### ISSUE TITLE
About bumping pikepdf version requirement

### ISSUE THREAD

**Issue by jeromerobert** [pdfarranger/pdfarranger#460]
> We may want to consider updating the dependency on the pikepdf version.

> I think requiring pikepdf >- 2.8.0 might be a good idea

Unfortunately, when packaged for Linux distros, requiring something which is too new is a problem because the pikepdf package may not be available in a sufficient version.

Repology seems to be a good source for more information on packaged versions: https://repology.org/project/python:pikepdf/versions

I once created this version Matrix with repology, maybe it can help with the decision-making - editing the URL should be quite self-explanatory:

https://repology.org/badge/versions-matrix.svg?projects=pdfarranger,python:pikepdf%3E=1.7,pdfshuffler&header=pdfarranger%20packaging%20status

I know that the pikepdf-Fedora-package-maintainers won't update to 2.x during a release because they consider this a major release (It is according to [Semantic Versioning](https://semver.org/lang/de/) because it goes from 1.x to 2.x, not sure if that is actually a breaking change in pikepdf that would prevent upgrading.) At any rate, the fallout should be carefully considered before raising the required versions. Probably in its own issue for better clarity and structure.

Edit: I could try discussing and fixing this for Fedora with the pikepdf maintainers. However, there are many distributions and I expect them to have very similar rules and problems.

_Originally posted by @dreua in https://github.com/pdfarranger/pdfarranger/pull/459#issuecomment-815672525_

**Comment by jeromerobert** [comment]
From Repology I would says that `>=1.17` would be a good trade off.

**Comment by dreua** [comment]
For comparison: Our current requirement in `setup.py` is `pikepdf>=1.7.0`

I think `>=1.17` should be no problem, `>=1.17.3`  should be fine, too, not sure whether it makes a difference for us.

**Comment by m-holger** [comment]
Perhaps another point to consider

```
v2.10.0

-    Fixed a XML External Entity (XXE) processing vulnerability in PDF XMP metadata parsing. 
     (Reported by Eric Therond of  Sonarsource.) 
     All users should upgrade to get this security update. CVE-2021-29421 was assigned to this issue.
```

**Comment by dreua** [comment]
Wrong audience: Please report this to the pikepdf maintainers of all the distros out there! We are not forcing anyone to use outdated pikepdf, PDF Arranger is, afaik, fully compatible with the latest pikepdf. (I'll update the flatpak asap, thanks for the reminder!)

**Comment by m-holger** [comment]
@dreua

>  are not forcing anyone to use outdated pikepdf

Putting on my user hat, I am not sure this works. I installed PDF Arranger twice this morning, once using the mintinstall software manager and once using pip. Both times I installed an application without any errors , on a system with all security updates applied. Both times I ended up with an application with a known vulnerability.

From a user perspective - I am a user of PDF Arranger, PDF Arranger is the user of pikepdf.
I have kept up with my updates of PDF Arranger and therefore have done my bit. I expect PDF Arranger to do its bit.

To me, PDF Arranger doing its bit would either mean requiring pikepdf>=2.10.0 or asking the developers of pikepdf to issue a security release of 1.x 

>  Please report this to the pikepdf maintainers of all the distros out there

I wouldn't have the slightest idea of how to do that. In any case, I assume that @jbarlow83 has already done this.
